apis/kueue/v1alpha1/tas_types.go

```diff
@@ -63,6 +63,11 @@ const (
 	// PodGroupPodIndexLabel is a label set on the Pod's metadata belonging
 	// to a Pod group. It indicates the Pod's index within the group.
 	PodGroupPodIndexLabel = "kueue.x-k8s.io/pod-group-pod-index"
+
+	// PodGroupPodIndexLabelNameFromAnnotation is an annotation on the Pod's metadata
+	// belonging to a Pod group. It indicates a label name used to retrieve the Pod's
+	// index within the group.
+	PodGroupPodIndexLabelNameFromAnnotation = "kueue.x-k8s.io/pod-group-pod-index-label-name-from"
```

Comment:
```suggestion
	PodGroupPodIndexLabelAnnotation = "kueue.x-k8s.io/pod-group-pod-index-label"
```
wdyt?

Reply:
Done

Reply:
I also suggested to rename the annotation itself `kueue.x-k8s.io/pod-group-pod-index-label-name-from` -> `kueue.x-k8s.io/pod-group-pod-index-label`

Reply:
Done